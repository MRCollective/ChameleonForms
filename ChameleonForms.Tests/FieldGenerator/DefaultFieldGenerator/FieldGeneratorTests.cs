using System.Web;
using ApprovalTests.Html;
using ChameleonForms.Component.Config;
using NUnit.Framework;

namespace ChameleonForms.Tests.FieldGenerator.DefaultFieldGenerator
{
    class FieldGeneratorTests : DefaultFieldGeneratorShould
    {
        [Test]
        public void Use_overridden_html_for_field_if_provided()
        {
            var g = Arrange(m => m.RequiredString);
            var field = new HtmlString("<p>override</p>");

            var result = g.GetFieldHtml(new FieldConfiguration().OverrideFieldHtml(field));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_without_config()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(default(IFieldConfiguration));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_class_for_field_label_with_html_attributes_including_a_class()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().AddClass("asdf").Disabled().Readonly());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_null_label()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_display_name()
        {
            var g = Arrange(m => m.StringWithDisplayAttribute);

            var result = g.GetLabelHtml(new FieldConfiguration());

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_overridden_label()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().Label("asdf"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_overridden_html_label()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().Label(new HtmlString("<em>lay</em>bell")));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_overridden_id()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().Attr(id => "DifferentId"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_correct_html_for_field_label_with_overridden_id_2()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().Id("DifferentId"));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }

        [Test]
        public void Use_label_text_when_without_label()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetLabelHtml(new FieldConfiguration().WithoutLabel());

            Assert.That(result.ToHtmlString(), Is.EqualTo("RequiredString"));
        }

        [Test]
        public void Use_correct_html_for_field_validation()
        {
            var g = Arrange(m => m.RequiredString);

            var result = g.GetValidationHtml(default(IFieldConfiguration));

            HtmlApprovals.VerifyHtml(result.ToHtmlString());
        }
    }
}
