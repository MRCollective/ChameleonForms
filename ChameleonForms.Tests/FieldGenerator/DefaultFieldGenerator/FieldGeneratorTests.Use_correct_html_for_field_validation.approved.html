<span class="field-validation-error" data-valmsg-for="RequiredString" data-valmsg-replace="true">asdf</span>